doc/source/development/contributing.rst

```diff
@@ -186,216 +186,6 @@ Note that you might need to rebuild the C extensions if/when you merge with upst
 
     python setup.py build_ext -j 4
 
-.. _contributing.dev_c:
```

Comment:
Rather than just removing this section, I'd suggest to put a link to the "contributing outside of docker" page

Reply:
Actually, if we just have a separate "setting up a development environment" page, then can just be a single link to that

Reply:
@dsaxton thoughts on this? It just seems a bit strange to me to special-case Docker

Reply:
@MarcoGorelli I think that makes sense. Something like `contributing_environment.rst` which includes _both_ Docker instructions as well as non-Docker, rather than `env_outside_docker.rst` and then still putting Docker environment instructions in main `contributing.rst`?